### PR TITLE
Ghmeding/express js base files v2

### PR DESCRIPTION
**Pull request by Ghmeding**
### What
This is the first PR for the Minitwit-service in Express.js.
It includes the basic web-application as well as the basic API-implementation.
I also added the tests that the other team created.

### Please note
To run, see _ExpressJSReadMe_

### CODE REVIEW

File: ExpressJS/utils/sequilize.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  sequelize = new Sequelize(
+    `postgres://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/waect`,
+    { // Sequlieze connection for postgres
+      logging: console.log
+  });
+  console.log(`postgres://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/waect`)
+
+} else if (dbConnectionType === 'prod') {
+
+  const ca_file = fs.readFileSync('/express-docker/secrets/ca-certificate.crt');
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:24]: we dont need this for now but whatever, we change it later 🥱

File: ExpressJS/utils/sequilize.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+console.log('POSTGRES_USER:', process.env.POSTGRES_USER);
+console.log('POSTGRES_PASSWORD:', process.env.POSTGRES_PASSWORD);
+console.log('POSTGRES_HOST:', process.env.POSTGRES_HOST);
+console.log('POSTGRES_PORT:', process.env.POSTGRES_PORT);
+console.log('DB_CONNECTION_TYPE:', process.env.DB_CONNECTION_TYPE);
+
+let sequelize;
+if (dbConnectionType === 'dev_db') {
+  sequelize = new Sequelize(
+    `postgres://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/waect`,
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:16]: you can use 
```suggestion
    `postgres://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/waect`,
```
```suggestion
    process.env.DATABASE_URL
```

File: ExpressJS/utils/sequilize.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
+require('dotenv').config({ path: './utils/dev.env' });
+const fs = require('fs');
+const { Sequelize, Op, Model, DataTypes } = require('sequelize');
+
+
+const dbConnectionType = process.env.DB_CONNECTION_TYPE;
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:6]: fuck this console.log

File: ExpressJS/utils/sequilize.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+// A user can be followed by many users
+Users.belongsToMany(Users, {
+  through: Followers,
+  as: 'UserFollowers',     // Users that follow this user
+  foreignKey: 'whom_id',
+  otherKey: 'who_id'
+});
+// Test database operations
+async function testDatabase() {
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:123]: Is this function testing the schema?? If is the case we can delete the init_db from the db.js

File: ExpressJS/utils/db.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+const { Sequelize } = require('sequelize');
+
+
+// PostgreSQL
+let pool = new Pool();
+const SCHEMA_FILE_PATH = 'src/utils/schema_postgres.sql';
+const dbConnectionType = process.env.DB_CONNECTION_TYPE;
+if (dbConnectionType === 'dev_db') {
+  pool = new Pool({
+    user: process.env.POSTGRES_USER,
```
Comment (dudustri) [inline ExpressJS/utils/db.js:15]: Is it possible to just pass the URL?? Then we don't need to have this amount of dotenv variable 🦘

File: ExpressJS/utils/sequilize.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
+require('dotenv').config({ path: './utils/dev.env' });
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:1]: Do we actually need to pass the path again here if it will be the same .env of the root folder?

**Review by dudustri — CHANGES_REQUESTED** [review]
Nice job. Most of the things are just comments we need to track for the refactor. However, I would change a few things on how the env variables are being used before merging it. If you disagree just comment your points and I approve it in the next review🏎️

File: ExpressJS/API/package.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "minitwit-2",
+    "version": "0.0.0",
+    "private": true,
+    "scripts": {
+      "start-api": "node --env-file='../utils/.env' ./src/api.js"
```
[the remark this answers is not in the document]
Comment (Ghmeding) [reply ExpressJS/API/package.json:6]: Think we can delete this actually. We can just load the env vars with dotenv

File: ExpressJS/ExpressJSReadMe.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+/API is the api
+
+### Dependencies
+node.js
+postgresql
+
+## How to run
+#### Local development
+1. Run 'npm install in root folder'
+2. Populate ./utils/.env
```
[the remark this answers is not in the document]
Comment (Ghmeding) [reply ExpressJS/ExpressJSReadMe.md:13]: Good idea

File: ExpressJS/utils/sequilize.js (hunk 3 above)
Comment (Ghmeding) [reply ExpressJS/utils/sequilize.js:6]: I guess are referring to the ones which logs the env-vars?

File: ExpressJS/utils/db.js (hunk 5 above)
Comment (Ghmeding) [reply ExpressJS/utils/db.js:15]: Not sure. Let's see. Maybe that's what causing the issue at your new branch

File: ExpressJS/utils/sequilize.js (hunk 3 above)
Comment (dudustri) [reply ExpressJS/utils/sequilize.js:6]: Sorry about the words xD
What I mean is to just delete the console log of the env variables

File: ExpressJS/utils/sequilize.js [hunk 9]
```diff
@@ -1,24 +1,15 @@
...
-console.log('POSTGRES_PORT:', process.env.POSTGRES_PORT);
-console.log('DB_CONNECTION_TYPE:', process.env.DB_CONNECTION_TYPE);
 
 let sequelize;
 if (dbConnectionType === 'dev_db') {
-  sequelize = new Sequelize(
-    `postgres://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/waect`,
-    { // Sequlieze connection for postgres
-      logging: console.log
+  sequelize = new Sequelize('waect', 'user', 'pass', {
```
Comment (dudustri) [inline ExpressJS/utils/sequilize.js:8]: Check my suggestion in the other PR.
you can just pass the URL here and change it in the .env before run

File: database/docker-compose.yml [hunk 10]
```diff
@@ -6,6 +6,8 @@ services:
       POSTGRES_DB: waect
       POSTGRES_USER: user
       POSTGRES_PASSWORD: pass
+      DB_HOST: host.docker.internals
```
Comment (dudustri) [inline database/docker-compose.yml:9]: Is this part of how you fixed the problem?

File: ExpressJS/latest_processed_sim_action_id.txt
Comment (dudustri) [inline ExpressJS/latest_processed_sim_action_id.txt:1]: do we need this to run?

File: .gitignore
Comment (dudustri) [inline .gitignore:1]: We can discuss about it. But it is more organized and clean to keep one gitignore per service + 1 in root instead a huge one in the root only.

File: .env.local
Comment (dudustri) [inline .env.local:1]: you don't need to delete this. There is no sensible content here.

File: ExpressJS/API/Dockerfile
Comment (dudustri) [inline ExpressJS/API/Dockerfile:1]: Why are you deleting the dockerfiles?

**Review by dudustri — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (Ghmeding) [reply .gitignore:1]: ok i will change it as last thing

File: ExpressJS/utils/sequilize.js (hunk 9 above)
Comment (dudustri) [reply ExpressJS/utils/sequilize.js:8]: > Then we can use the same .env! It is nice if we could keep only one env var for the db conneciton

File: database/docker-compose.yml (hunk 10 above)
Comment (Ghmeding) [reply database/docker-compose.yml:9]: yes

File: ExpressJS/API/Dockerfile
Comment (Ghmeding) [reply ExpressJS/API/Dockerfile:1]: did not use them and probably will make new

File: ExpressJS/latest_processed_sim_action_id.txt
Comment (Ghmeding) [reply ExpressJS/latest_processed_sim_action_id.txt:1]: yes

File: ExpressJS/utils/sequilize.js (hunk 9 above)
Comment (Ghmeding) [reply ExpressJS/utils/sequilize.js]: done

File: .gitignore
Comment (dudustri) [inline .gitignore:1]: can you clean up the root gitignore?

File: .env.local
Comment (dudustri) [inline .env.local:1]: Why did you deleted this one?

**Review by dudustri — COMMENTED** [review]
just clean the gitignore and maybe restore the .env.local and you are good to merge

**Pull request merged**